source/loader/ur_adapter_registry.hpp

```diff
@@ -88,9 +83,56 @@ class AdapterRegistry {
     static constexpr std::array<const char *, 1> knownPlatformNames{
         MAKE_LIBRARY_NAME("ur_adapter_level_zero", "0")};
 
+    template <typename T, typename = std::enable_if_t<
+                              std::is_constructible_v<T, const std::string &>>>
+    std::vector<T> getEnvVarValuesToVec(const std::string envVar,
+                                        const char separator) {
+        std::optional<std::string> envString;
+
+        try {
+            envString = ur_getenv(envVar.c_str());
+        } catch (const std::invalid_argument &e) {
+            logger::error(e.what());
+        }
+
+        std::vector<T> values;
+        if (envString) {
+            std::stringstream ss(*envString);
+            while (ss.good()) {
+                std::string value;
+                getline(ss, value, separator);
+                values.emplace_back(value);
+            }
+        }
+
+        return values;
+    }
+
+    std::vector<fs::path> getEnvAdapterSearchPaths() {
+        return getEnvVarValuesToVec<fs::path>("UR_ADAPTERS_SEARCH_PATH", ',');
+    }
+
     void discoverKnownAdapters() {
-        for (const auto &path : knownPlatformNames) {
-            discovered_adapters.emplace_back(path);
+        auto searchPaths = getEnvAdapterSearchPaths();
+        auto loaderPath = getLoaderLibDirectory();
```

Comment:
Would this result in adapter libraries in loader lib directory to take precedence over libraries discovered in LD_LIBRARY_PATH?

As a user I would expect LD_LIBRARY_PATH to have highest search precedence

Reply:
I've set the highest search precedence to `UR_ADAPTERS_SEARCH_PATH` and made `LD_LIBRARY_PATH` a higher priority than the current loader lib directory.